packages/x-components/src/x-modules/url/store/actions/update-url.action.ts

```diff
@@ -0,0 +1,34 @@
+import { forEach } from '../../../../utils/object';
+import { UrlXStoreModule } from '../types';
+
+/**
+ * Default implementation for the {@link UrlActions.updateUrl}.
+ *
+ * @param context - The {@link https://vuex.vuejs.org/guide/actions.html | context} of the actions,
+ * provided by Vuex.
+ *
+ * @internal
+ */
+export const updateUrl: UrlXStoreModule['actions']['updateUrl'] = ({
+  getters: { urlParams },
+  state: {
+    config: { urlParamNames }
+  }
+}) => {
+  const url = new URL(window.location.href);
+
+  forEach(urlParams, (paramKey, paramValue) => {
+    const newKey = urlParamNames[paramKey] ?? paramKey;
+    url.searchParams.delete(newKey);
```

Comment:
Maybe we can have a new getter that normalises the url param names, so instead of doing `const newKey = urlParamNames[paramKey] ?? paramKey;` you just do `const newKey = getters.urlParamNames[paramKey]`, and use that getter to clean the parameters?